python/lsst/ctrl/oods/fileIngester.py

```diff
@@ -217,67 +182,48 @@ def create_link_to_file(self, filename, dirname):
 
         return new_file
 
-    def stageFiles(self, msg):
-        """ stage the files to their butler staging areas
-        and remove the original file
-
-        Parameters
-        ----------
-        msg: `dict`
-            message structure
+    def stageFiles(self, file_list):
+        """Stage all files from the initial directory to directories
+        specific to each butler.
         """
-        filename = os.path.realpath(msg['FILENAME'])
-
-        try:
-            for butlerProxy in self.butlers:
-                local_staging_dir = butlerProxy.getStagingDirectory()
-                self.create_link_to_file(filename, local_staging_dir)
-        except Exception:
-            LOGGER.info(f"error staging files butler for {filename}")
-            return
-        # file has been linked to all staging areas;
-        # now we unlink the original file.
-        os.unlink(filename)
-
-    async def ingest(self, msg):
+        files = {}
+        for butlerProxy in self.butlers:
+            files[butlerProxy] = []
+        for filename in file_list:
+            try:
+                for butlerProxy in self.butlers:
+                    local_staging_dir = butlerProxy.getStagingDirectory()
+                    newfile = self.create_link_to_file(filename, local_staging_dir)
+                    files[butlerProxy].append(newfile)
+            except Exception:
+                LOGGER.info(f"error staging files butler for {filename}")
```

Comment:
In a future update please don't use f-strings in log messages. This should be written as:

```python
LOGGER.info("error staging files butler for %s", filename)
```
The aim being to delay the string interpolation until it's needed.